content/en/docs/concepts/storage/projected-volumes.md

```diff
@@ -71,18 +71,24 @@ A container using a projected volume source as a [`subPath`](/docs/concepts/stor
 volume mount will not receive updates for those volume sources.
 {{< /note >}}
 
-## SecurityContext interactions
+### serviceAccountToken and securityContext
 
 The [proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2451-service-account-token-volumes#proposal) for file permission handling in projected service account volume enhancement introduced the projected files having the the correct owner permissions set.
 
-### Linux
+#### Linux
 
-In Linux pods that have a projected volume and `RunAsUser` set in the Pod
-[`SecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context),
-the projected files have the correct ownership set including container user
-ownership.
+Kubernetes will improve security for serviceAccountToken files when all containers
+have the same `runAsUser` set in
+[`PodSecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context)
+or container
+[`SecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-1)
+by setting the file owner to the `runAsUser` and the file mode to `0600`.
```

Comment:
```suggestion
In some cases, Kubernetes applies a security optimization for the contents of`serviceAccountToken`
volumes.
When all containers in a pod have the same `runAsUser` set in their
[`PodSecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context)
or container
[`SecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-1),
then the kubelet  ensures that the contents of the `serviceAccountToken` volume are owned by that user,
and that every file has its permission mode set to `0600`.
```

(what about the gid of that file? is is set to 0)?

Reply:
👍 Much better, thanks. I was struggling to communicate this clearly.

Reply:
Oh, and I'd have to double check but I think the gid will be set to `fsGroup` if it's defined, root otherwise.